c/fasl.c

```diff
@@ -327,109 +312,59 @@ void S_fasl_init(void) {
 #endif
 }
 
-ptr S_fasl_read(INT fd, IFASLCODE situation, ptr path, ptr externals) {
-  ptr tc = get_thread_context();
-  ptr x; struct unbufFaslFileObj uffo;
+void S_fasl_init_fd(fileFaslFile ffo, ptr path, INT fd,
+                    int buffer_mode, uptr size) {
+  ffo->f.uf.path = path;
+  ffo->f.uf.type = UFFO_TYPE_FD;
+  ffo->f.uf.fd = fd;
 
-  uffo.path = path;
-  uffo.type = UFFO_TYPE_FD;
-  uffo.fd = fd;
-  x = fasl_entry(tc, situation, &uffo, externals);
-  return x;
+  ffo->f.buffer_mode = buffer_mode;
+  ffo->f.remaining = size;
+  ffo->f.next = ffo->f.end = ffo->f.buf = ffo->buf_space;
 }
 
-ptr S_bv_fasl_read(ptr bv, int ty, uptr offset, uptr len, ptr path, ptr externals) {
-  ptr tc = get_thread_context();
-  ptr x; struct unbufFaslFileObj uffo;
+void S_fasl_init_bytes(faslFile ffo, ptr path, void *data, iptr len) {
+  ffo->uf.path = path;
+  ffo->uf.type = UFFO_TYPE_BV;
+  ffo->uf.fd = 0;
```

Comment:
Should this be -1, since file descriptor 0 is STDIN?

Reply:
That's a good idea.